src/site/_drafts/what-should-you-measure-to-improve-performance/index.md

```diff
@@ -0,0 +1,175 @@
+
+### WebPageTest
+
+[WebPageTest](https://www.webpagetest.org/) offers a very flexible scripting
+solution described
+[here](https://sites.google.com/a/webpagetest.org/docs/using-webpagetest/scripting).
+The basic idea is to:
+
++   Tell WebPageTest to navigate through the pages of the flow with the
+    [`navigate`](https://sites.google.com/a/webpagetest.org/docs/using-webpagetest/scripting#TOC-navigate)
+    command.
++   If needed script the clicking of buttons via
+    [`clickAndWait`](https://sites.google.com/a/webpagetest.org/docs/using-webpagetest/scripting#TOC-clickAndWait)
+    commands and fill text fields via
+    [`setValue`](https://sites.google.com/a/webpagetest.org/docs/using-webpagetest/scripting#TOC-setValue).
+    For testing of Single Page Applications use `clickAndWait` rather than
+    `navigate` commands for all steps after the first, as `navigate` will do a
+    full load instead of the lightweight virtual page load.
++   Make sure to combine the different steps of the flow in the analysis via
+    [`combineSteps`](https://sites.google.com/a/webpagetest.org/docs/using-webpagetest/scripting#TOC-combineSteps)
+    to produce a single overall result report for the complete flow.
+
+Such a script could look like this:
+
+```
+combineSteps  
+navigate	https://www.store.google.com/landingpage  
+navigate	https://www.store.google.com/productpage
+clickAndWait	innerText=Buy Now  
+navigate	https://www.store.google.com/basket  
+navigate	https://www.store.google.com/checkout
+```
+
+With a script like this in place you can easily measure and compare performance
+over time. This can even be automated through the
+[WebPageTest API](https://sites.google.com/a/webpagetest.org/docs/advanced-features/webpagetest-restful-apis).
+
+### Puppeteer
+
+Another great option to script testing is via headless Chrome, which can be
+controlled through the Node API
+[Puppeteer](https://github.com/GoogleChrome/puppeteer). The general idea is to
+start the browser through Puppeteer, navigate to the landing page through the
+[goto](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pagegotourl-options)
+function,
+[inject Javascript](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pageevaluatepagefunction-args)
+to fill fields or
+[click](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pageclickselector-options)
+buttons and proceed through the funnel through further
+[goto](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pagegotourl-options)
+calls as needed. As a metric the duration of the flow can be measured directly,
+but you could also sum up the FCP, FMP or TTI values of the individual loads of
+the flow.
+[Test website performance with Puppeteer](https://michaljanaszek.com/blog/test-website-performance-with-puppeteer)
+provides an overview of how to get performance metrics via Puppeteer. A very
+simplified example node script could look like this:
+
+```
```

Comment:
```suggestion
```js
```

enable syntax highlighting

Reply:
Added in, but didn't test (as done through github UI as I'm low on time)